From 92eb3e9767cb8afb8fb222d37b63427bb603b5d8 Mon Sep 17 00:00:00 2001
From: Andreas Sturmlechner <andreas.sturmlechner@gmail.com>
Date: Wed, 31 Jan 2018 15:32:46 +0100
Subject: [PATCH] Make sure to search for Qt5-based qmlplugindump

Summary: Without any hint, qmlplugindump version is whatever default is set by qtchooser.

Test Plan: ecm_find_qmlmodule now works properly for e.g. kirigami.

Reviewers: apol, cgiboudeaux

Reviewed By: cgiboudeaux

Subscribers: aacid, dfaure, cgiboudeaux, #frameworks, #build_system

Tags: #frameworks, #build_system

Differential Revision: https://phabricator.kde.org/D9116
---
 modules/ECMFindQMLModule.cmake.in | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/modules/ECMFindQMLModule.cmake.in b/modules/ECMFindQMLModule.cmake.in
index 428d60a9..8109ab8f 100644
--- a/modules/ECMFindQMLModule.cmake.in
+++ b/modules/ECMFindQMLModule.cmake.in
@@ -26,8 +26,11 @@
 #=============================================================================
 
 include(FindPackageHandleStandardArgs)
+include("${ECM_MODULE_DIR}/ECMQueryQmake.cmake")
 
-find_program(QMLPLUGINDUMP_PROGRAM qmlplugindump)
+query_qmake(qt_binaries_dir QT_INSTALL_BINS)
+
+find_program(QMLPLUGINDUMP_PROGRAM NAMES qmlplugindump HINTS ${qt_binaries_dir})
 if(NOT QMLPLUGINDUMP_PROGRAM)
     message(WARNING "Could not find qmlplugindump. It is necessary to look up qml module dependencies.")
 endif()
